libs/errors/src/exception.cpp

```diff
@@ -276,6 +276,47 @@ namespace hpx { namespace detail {
     template HPX_EXPORT std::exception_ptr get_exception(hpx::exception const&,
         std::string const&, std::string const&, long, std::string const&);
 
+    template HPX_EXPORT std::exception_ptr get_exception(
+        boost::system::system_error const&, std::string const&,
+        std::string const&, long, std::string const&);
+
+    template HPX_EXPORT std::exception_ptr get_exception(std::exception const&,
+        std::string const&, std::string const&, long, std::string const&);
+    template HPX_EXPORT std::exception_ptr get_exception(
+        hpx::detail::std_exception const&, std::string const&,
+        std::string const&, long, std::string const&);
+    template HPX_EXPORT std::exception_ptr get_exception(
+        std::bad_exception const&, std::string const&, std::string const&, long,
+        std::string const&);
+    template HPX_EXPORT std::exception_ptr get_exception(
+        hpx::detail::bad_exception const&, std::string const&,
+        std::string const&, long, std::string const&);
+    template HPX_EXPORT std::exception_ptr get_exception(
+        hpx::detail::bad_typeid const&, std::string const&, std::string const&,
+        long, std::string const&);
+    template HPX_EXPORT std::exception_ptr get_exception(std::bad_cast const&,
+        std::string const&, std::string const&, long, std::string const&);
+    template HPX_EXPORT std::exception_ptr get_exception(
+        hpx::detail::bad_cast const&, std::string const&, std::string const&,
+        long, std::string const&);
+    template HPX_EXPORT std::exception_ptr get_exception(std::bad_alloc const&,
+        std::string const&, std::string const&, long, std::string const&);
+    template HPX_EXPORT std::exception_ptr get_exception(
+        hpx::detail::bad_alloc const&, std::string const&, std::string const&,
+        long, std::string const&);
+    template HPX_EXPORT std::exception_ptr get_exception(
+        std::logic_error const&, std::string const&, std::string const&, long,
+        std::string const&);
+    template HPX_EXPORT std::exception_ptr get_exception(
+        std::runtime_error const&, std::string const&, std::string const&, long,
+        std::string const&);
+    template HPX_EXPORT std::exception_ptr get_exception(
+        std::out_of_range const&, std::string const&, std::string const&, long,
+        std::string const&);
+    template HPX_EXPORT std::exception_ptr get_exception(
+        std::invalid_argument const&, std::string const&, std::string const&,
+        long, std::string const&);
```

Comment:
Why do we need these explicit instantiations? I think these will be instantiated implicitly by the explicit instantiations of the `throw_exception` functions below.

Reply:
They are not instantiated otherwise. I did not dig into why though.

Reply:
Interesting - hmmm. They were instantiated before, did you change something?

Reply:
I'll try removing them again. Maybe they were only needed in an earlier iteration of this PR. There are not so many changes here so I should be able to figure out why they're not instantiated (if they indeed aren't).

Reply:
It was indeed not needed anymore. Thanks for pointing this out!